src/qute/commands/registerCommands.ts

```diff
@@ -95,6 +116,15 @@ export function registerConfigurationUpdateCommand(context: ExtensionContext) {
   }));
 }
 
+/**
+   * Sets the `quteLanguageSupported` context to `true` if the editor language id
+   * is a qute template. Sets to `false` otherwise.
+   */
+ export async function checkQuteLanguageSupportedContext() {
+  const isSupportedOpen = window.activeTextEditor.document.languageId.includes('qute')
```

Comment:
Would it be possible to pass in the `document` as a parameter here from the `onDidOpenTextDocument` call. It would avoid having to fetch it through the UI element. It shouldn't fail this way, but passing it in seems safer.

Reply:
It seems when using the `document` param from `onDidOpenTextDocument`, the button will appear/disappear each time a file is opened. It seems each time a text document is opened, the first run of the listener returns true for `checkQuteLanguageContext`, then it will run again and return false and the button will disappear.

Reply:
And you added `await` to the `updateLanguageId` call right ? Still doesn't work ?

Reply:
And you added `await` to the `updateLanguageId` call right ? Still doesn't work ?

Reply:
Yes, same result:
![Peek 2022-02-03 13-42](https://user-images.githubusercontent.com/26389510/152408024-24bc8811-5885-4050-b32e-a1a660bfc731.gif)

Reply:
Ok I think I know why this is happening.

I just added a breakpoint on `isSupportedOpen` of `checkQuteLanguageSupportedContext`. The callback `onDidOpenTextDocument` is triggered twice. Once for the file, and a second time on the same file with `.git` as suffix. I've seen this before, but no idea why the extra event.

Let's leave it as-is and maybe I can investigate why that happens.

Reply:
See https://github.com/microsoft/vscode/issues/131282 . Probably some git capabilities are generating the extra file. I guess we could technically have a check to ignore documents where the uri scheme is git.